rom/laghos.cpp

```diff
@@ -394,8 +394,15 @@ int main(int argc, char *argv[])
     romOptions.basisIdentifier = std::string(basisIdentifier);
 
     romOptions.hyperreductionSamplingType = getHyperreductionSamplingType(hyperreductionSamplingType);
-    romOptions.use_sample_mesh = romOptions.hyperreduce && (romOptions.hyperreductionSamplingType != eqp);
-    MFEM_VERIFY(!(romOptions.SNS) || (romOptions.hyperreductionSamplingType != eqp), "Using SNS with EQP is prohibited");
+    romOptions.use_sample_mesh = romOptions.hyperreduce && (romOptions.hyperreductionSamplingType != eqp
+			|| romOptions.hyperreductionSamplingType != eqp_energy);
+
+    MFEM_VERIFY(!(romOptions.SNS) || (romOptions.hyperreductionSamplingType != eqp &&
+			romOptions.hyperreductionSamplingType != eqp_energy),
+			"Using SNS with EQP is prohibited");
+	
+	if (romOptions.hyperreduce && romOptions.hyperreductionSamplingType == eqp_energy)
+		romOptions.GramSchmidt = true;
```

Comment:
I added this check here to ensure that Gram-Schmidt is performed when using the energy-conserving EQP. 

My understanding is that by setting this to true, we are normalizing the basis matrices so that the reduced mass matrices are identities. Is that correct?

Reply:
That's correct. This will do the Gram Schmidt no matter what the user-defined option is. Although we may want to use Gram Schmidt for most of our runs, we may still want to keep the option alive while it exists. 

I'm a little concerned about whether the Gram Schmidt process is still functioning. We have been using SNS and not using Gram Schmidt for a while.